source/m3_config_platforms.h

```diff
@@ -161,10 +161,27 @@ typedef short           i16;
 typedef uint8_t         u8;
 typedef int8_t          i8;
 
+static inline float rintf( float arg ) {
+  union { float f; uint32_t i; } u;
+  u.f = arg;
+  uint32_t ux = u.i & 0x7FFFFFFF;
+  if (UNLIKELY(ux == 0 || ux > 0x5A000000)) {
+    return arg;
+  }
+  return (float)lrint(arg);
+}
+
+static inline double rint( double arg ) {
+  union { double f; uint32_t i[2]; } u;
+  u.f = arg;
+  uint32_t ux = u.i[1] & 0x7FFFFFFF;
```

Comment:
@vshymanskyy I'm not really familiar with Arduino but it seems it uses little endian. Overwise it will incorrect

Reply:
It looks like really strange with `u.i[1]`: https://godbolt.org/z/eGJRZT
Just get rid of everything...

Reply:
Alright. It seems I fixed this: https://godbolt.org/z/Wgi-e2